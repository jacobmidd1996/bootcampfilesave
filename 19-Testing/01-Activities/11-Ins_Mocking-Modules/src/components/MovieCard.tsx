import MovieDetails from './MovieDetails'

export interface MovieProps {
  movie: {
    title: string;
    description: string;
    yearReleased: number;
    IMDbRating: number;
  };
}

export default function MovieCard({ movie }: MovieProps) {
  return (
    <div className="col-md-4">
      <div className="card h-100">
        <div className="card-body">
          <h3 className="card-title">{movie.title}</h3>
          <p className="card-text">{movie.description}</p>
          <MovieDetails movie={movie} />
        </div>
      </div>
    </div>
  );
}
