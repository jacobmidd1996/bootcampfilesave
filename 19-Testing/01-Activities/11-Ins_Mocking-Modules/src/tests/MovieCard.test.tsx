import { render } from "@testing-library/react";
import MovieCard from "../components/MovieCard";
import pretty from 'pretty';
import { vi } from 'vitest'

const movie = {
  title: "Longlegs",
  description: "In pursuit of a serial killer, an FBI agent uncovers a series of occult clues that she must solve to end his terrifying killing spree.",
  yearReleased: 2024,
  IMDbRating: 7.2,
}

vi.mock('../components/MovieDetails', () => {
  return {
    default: () => <div>
      <p>Year Released: 2023</p>
      <p>IMDb Rating: 9.9</p>
    </div>
  }
});

describe('MovieCard', () => {
  it('should render a MovieCard with the mocked MovieDetails component', () => {
    render(<MovieCard movie={movie} />);

    const movieCardTitle = document.querySelector('h3');
    const movieDetails = document.querySelectorAll('p');

    const description = movieDetails[0].textContent;
    const yearReleased = movieDetails[1].textContent;
    const imdbRating = movieDetails[2].textContent;

    expect(movieCardTitle?.textContent).toBe('Longlegs');
    expect(description).toBe('In pursuit of a serial killer, an FBI agent uncovers a series of occult clues that she must solve to end his terrifying killing spree.');
    expect(yearReleased).toBe('Year Released: 2023');
    expect(imdbRating).toBe('IMDb Rating: 9.9');
  });

  it('should render a MovieCard that matches the snapshot', () => {
    render(<MovieCard movie={movie} />);
    
    const movieCard = document.querySelector('.card');

    if(movieCard) {
      expect(pretty(movieCard.innerHTML)).toMatchSnapshot();
    }
  });
});
