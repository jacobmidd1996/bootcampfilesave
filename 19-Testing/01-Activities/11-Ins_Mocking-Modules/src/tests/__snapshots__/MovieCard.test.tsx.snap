// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MovieCard > should render a MovieCard that matches the snapshot 1`] = `
"<div class="card-body">
  <h3 class="card-title">Longlegs</h3>
  <p class="card-text">In pursuit of a serial killer, an FBI agent uncovers a series of occult clues that she must solve to end his terrifying killing spree.</p>
  <div>
    <p>Year Released: 2023</p>
    <p>IMDb Rating: 9.9</p>
  </div>
</div>"
`;
